File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/responses/MemberResponseDto.java
package com.jetbrains.gymusserverjava.memberships.dtos.responses;

import java.time.LocalDate;
import java.time.LocalDateTime;

public record MemberResponseDto(
        int id,
        String firstName,
        String lastName,
        String email,
        String phoneNumber,
        String address,
        LocalDate birthdate,
        String personalPhoto,
        LocalDate endDate,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    public boolean isActive() {
        return endDate.isAfter(LocalDate.now());
    }

}

File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/responses/MemberResponseDto.kt
package com.jetbrains.gymusserverjava.memberships.dtos.responses

import java.time.LocalDate
import java.time.LocalDateTime

data class MemberResponseDto(
    val id: Int,
    val firstName: String,
    val lastName: String,
    val email: String,
    val phoneNumber: String,
    val address: String,
    val birthdate: LocalDate,
    val personalPhoto: String,
    val endDate: LocalDate,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
) {
    val isActive: Boolean
        get() = endDate.isAfter(LocalDate.now())
}
